### PR TITLE
Add action to spell check docs and README

### PR DESCRIPTION
**Pull request by Westwooo**
Addresses: https://github.com/couchbaselabs/couchbase-shell/issues/442

### CODE REVIEW

File: .github/workflows/.spellcheck.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+matrix:
+  - name: Markdown
+    expect_match: false
+    apsell:
+      mode: en
```
[the remark this answers is not in the document]
Comment (Westwooo) [reply .github/workflows/.spellcheck.yml]: The commands specifically use American english, e.g neighbors in vector search, so the docs should be the same.

File: .github/workflows/.spellcheck.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            - alt
+          ignores:
+            - ':matches(code, pre)'
+            - 'code'
+            - 'pre'
+            - 'blockquote'
+    sources:
+      - 'README.md'
+      - 'docs/*.adoc'
+      - 'docs/**/*.adoc'
```
[the remark this answers is not in the document]
Comment (Westwooo) [reply .github/workflows/.spellcheck.yml]: Done!

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -159,4 +159,15 @@ jobs:
...
+  check-spelling:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Check Spelling
+        uses: rojopolis/spellcheck-github-actions@0.23.0
+        with:
+          config_path: .github/workflows/.spellcheck.yml
+          task_name: Markdown
```
[the remark this answers is not in the document]
Comment (Westwooo) [reply .github/workflows/ci.yml]: Done!

File: .github/workflows/.spellcheck.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Peter-Searby) [reply .github/workflows/.spellcheck.yml]: Makes sense

**Review by Peter-Searby — APPROVED** [review]

**Pull request merged**